File: app/src/main/java/com/klim/us_stock/di/search/SearchComponent.java
package com.klim.us_stock.di.search;

import com.klim.us_stock.ui.windows.search.SearchFragment;

import dagger.Subcomponent;

@SearchScope
@Subcomponent(modules = {SearchModule.class})
public interface SearchComponent {

    public void inject(SearchFragment fragment);

}

File: src/app/src/main/java/com/klim/us_stock/di/search/SearchComponent.kt
package com.klim.us_stock.di.search

import com.klim.dep_in.ApplicationContextProvider
import com.klim.us_stock.ui.windows.search.SearchFragment
import dagger.Component

@SearchScope
@Component(
    dependencies = [ApplicationContextProvider::class],
    modules = [SearchModule::class]
)
interface SearchComponent {
    fun inject(fragment: SearchFragment?)

    class Initializer private constructor() {
        companion object {
            fun init(appComponent: ApplicationContextProvider): SearchComponent =
                DaggerSearchComponent.builder()
                    .applicationContextProvider(appComponent)
                    .build()
        }
    }
}